src/main/kotlin/no/skatteetaten/aurora/databasehotel/dao/oracle/OracleDatabaseHotelDataDao.kt

```diff
@@ -0,0 +1,233 @@
+package no.skatteetaten.aurora.databasehotel.dao.oracle
+
+import com.google.common.collect.Lists.newArrayList
+import no.skatteetaten.aurora.databasehotel.dao.DataAccessException
+import no.skatteetaten.aurora.databasehotel.dao.DatabaseHotelDataDao
+import no.skatteetaten.aurora.databasehotel.dao.DatabaseSupport
+import no.skatteetaten.aurora.databasehotel.dao.SchemaTypes.SCHEMA_TYPE_MANAGED
+import no.skatteetaten.aurora.databasehotel.dao.dto.ExternalSchema
+import no.skatteetaten.aurora.databasehotel.dao.dto.Label
+import no.skatteetaten.aurora.databasehotel.dao.dto.SchemaData
+import no.skatteetaten.aurora.databasehotel.dao.dto.SchemaUser
+import org.springframework.jdbc.core.BeanPropertyRowMapper
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import java.util.Collections
+import java.util.Date
+import java.util.Optional
+import java.util.UUID
+import javax.sql.DataSource
+
+open class OracleDatabaseHotelDataDao(dataSource: DataSource) : DatabaseSupport(dataSource), DatabaseHotelDataDao {
+
+    private fun generateId(): String {
+
+        return UUID.randomUUID().toString()
+    }
+
+    override fun createSchemaData(name: String): SchemaData {
+
+        return createSchemaData(name, SCHEMA_TYPE_MANAGED)
+    }
+
+    override fun createSchemaData(name: String, schemaType: String): SchemaData {
+
+        val id = generateId()
+        jdbcTemplate
+            .update("insert into SCHEMA_DATA (id, name, schema_type) values (?, ?, ?)", id, name, schemaType)
+        return Optional.ofNullable(findSchemaDataById(id))
```

Comment:
Erstatte med noe ala

`return findSchemaDataById(id) ?: throw DataAccessException("Unable to create schema data")`